app/userOperations.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Created by Sahil Jain on 20/08/2014.
+ */
+var Question = require('../models/question');
+var Answer = require('../models/answer');
+var textUtils = require('../utils/textutils');
+var tokenUtils = require('../utils/tokenutils');
+var Constants = require('../constants');
+
+var serverError = function (res) {
+    console.log("sending error");
+    return res.status(500).json({error: "server error"});
+};
+
+module.exports = function (passport) {
+    return setupFunctions(passport);
+};
+
+var setupFunctions = function (passport) {
+    var exports = {};
+
+    exports.getUser = function (req, res) { //TODO format the output of questions/answers correctly
+        var token = req.query.token;
+        if (textUtils.isEmpty(token)) {
+            return res.status(400).json({error: Constants.ERROR.MISSING.TOKEN});
+        }
+
+        tokenUtils.getUserFromToken(token, function (err, user) {
```

Comment:
Promises will avoid situations like this where the function just keeps going deeper and deeper (more and more indentations)